src/erasure.rs

```diff
@@ -347,101 +348,44 @@ pub fn generate_coding(
     Ok(())
 }
 
-// examine the window slot at idx returns
-//  true if slot is empty
-//  true if slot is stale (i.e. has the wrong index), old blob is flushed
-//  false if slot has a blob with the right index
-fn is_missing(id: &Pubkey, idx: u64, window_slot: &mut Option<SharedBlob>, c_or_d: &str) -> bool {
-    if let Some(blob) = window_slot.take() {
-        let blob_idx = blob.read().unwrap().index().unwrap();
-        if blob_idx == idx {
-            trace!("recover {}: idx: {} good {}", id, idx, c_or_d);
-            // put it back
-            mem::replace(window_slot, Some(blob));
-            false
-        } else {
-            trace!(
-                "recover {}: idx: {} old {} {}, recycling",
-                id,
-                idx,
-                c_or_d,
-                blob_idx,
-            );
-            true
-        }
-    } else {
-        trace!("recover {}: idx: {} None {}", id, idx, c_or_d);
-        // nothing there
-        true
-    }
-}
-
-// examine the window beginning at block_start for missing or
-//  stale (based on block_start_idx) blobs
-// if a blob is stale, remove it from the window slot
-//  side effect: block will be cleaned of old blobs
-fn find_missing(
-    id: &Pubkey,
-    block_start_idx: u64,
-    block_start: usize,
-    window: &mut [WindowSlot],
-) -> (usize, usize) {
-    let mut data_missing = 0;
-    let mut coding_missing = 0;
-    let block_end = block_start + NUM_DATA;
-    let coding_start = block_start + NUM_DATA - NUM_CODING;
-
-    // count missing blobs in the block
-    for i in block_start..block_end {
-        let idx = (i - block_start) as u64 + block_start_idx;
-        let n = i % window.len();
-
-        if is_missing(id, idx, &mut window[n].data, "data") {
-            data_missing += 1;
-        }
-
-        if i >= coding_start && is_missing(id, idx, &mut window[n].coding, "coding") {
-            coding_missing += 1;
-        }
-    }
-    (data_missing, coding_missing)
-}
-
 // Recover a missing block into window
 //   missing blocks should be None or old...
 //   If not enough coding or data blocks are present to restore
-//    any of the blocks, the block is skipped.
+//   any of the blocks, the block is skipped.
 //   Side effect: old blobs in a block are None'd
-pub fn recover(id: &Pubkey, window: &mut [WindowSlot], start_idx: u64, start: usize) -> Result<()> {
-    let block_start = start - (start % NUM_DATA);
+pub fn recover(
+    db_ledger: &mut DbLedger,
+    slot: u64,
+    start_idx: u64,
+) -> Result<(Vec<SharedBlob>, Vec<SharedBlob>)> {
```

Comment:
What is the result?

Reply:
also, the function comment doesn't seem accurate, it doesn't take a window anymore.

Reply:
The result is a list of reconstructed data and coding blobs returned to the caller for processing.

Reply:
Ok, yea just could be indicated with a comment or something, I think a reader of the the function wouldn't realize it just reading the signature.

Reply:
For sure, updated the comment